book/chapters/chapter9/preprocessing.qmd

```diff
@@ -244,26 +240,33 @@ bmr$aggregate(measure = msr_mae)[, .(learner_id, regr.mae)]
-2) `po("colapply")` -- Character and ordinal features are encoded as categorical, and date/time features are encoded as numeric.
-3) `po("imputehist")` -- Numeric features are imputed by histogram.
-4) `po("imputesample")` -- Logical features are imputed by sampling from the empirical distribution.
-5) `po("missind")` -- Missing data indicators are added for imputed numeric and logical variables
-6) `po("imputeoor")` -- Missing values of categorical features are encoded with a new level
-7) `po("fixfactors")` -- Fixes levels of categorical features such that the same levels are present during prediction and training (which may involve dropping empty factor levels)
-8) `po("imputesample")` -- Missing values in categorical features introduced from dropping levels in the previous step are imputed by sampling from the empirical distributions.
-9) `po("collapsefactors")` -- Categorical features levels are collapsed (starting from the rarest factors in the training data) until there are less than 1000 levels
-10) `po("encode")` -- Categorical features are one-hot encoded
-11) `po("removeconstants")` -- Constant features that might have been created in the previous steps are removed
+1. `po("removeconstants")` -- Constant features are removed.
+2. `po("colapply")` -- Character and ordinal features are encoded as categorical, and date/time features are encoded as numeric.
+3. `po("imputehist")` -- Numeric features are imputed by histogram sampling.
+4. `po("imputesample")` -- Logical features are imputed by sampling from the empirical distribution -- this only affects the `$predict()`-step.
+5. `po("missind")` -- Missing data indicators are added for imputed numeric and logical variables.
+6. `po("imputeoor")` -- Missing values of categorical features are encoded with a new level.
+7. `po("fixfactors")` -- Fixes levels of categorical features such that the same levels are present during prediction and training (which may involve dropping empty factor levels).
+8. `po("imputesample")` -- Missing values in categorical features introduced from dropping levels in the previous step are imputed by sampling from the empirical distributions.
+9. `po("collapsefactors")` -- Categorical features levels are collapsed (starting from the rarest factors in the training data) until there are less than a certan number of levels.
+  This number is controlled by the `max_cardinality` argument and defaults to a conservative 1000.
+10. `po("encode")` -- Categorical features are one-hot encoded.
+11. `po("removeconstants")` -- Constant features that might have been created in the previous steps are removed.
+
+`ppl("robustify")` has optional arguments `task` and `learner`.
+If these are provided, then the resulting pipeline will be set up to handle the given task and learner specifically: it will e.g. not impute missing values if the learner has the `"missings"` property, or if there are no missing values in the task to begin with.
```

Comment:
```suggestion
If these are provided, then the resulting pipeline will be set up to handle the given task and learner specifically, for example, it will not impute missing values if the learner has the `"missings"` property, or if there are no missing values in the task to begin with.
```